Data/ByteString/Builder/RealFloat/Internal.hs

```diff
@@ -735,12 +787,12 @@ packWord16 l h =
 #endif
 
 -- | Unpacks a 16-bit word into 2 bytes [lsb, msb]
-unpackWord16 :: Word# -> (# Word#, Word# #)
+unpackWord16 :: Word# -> (# Word8#, Word8# #)
```

Comment:
`Word8#` doesn't exist until ghc-8.8, and the conversion operators were renamed in ghc-9.2, with no compatibility wrappers left behind that I could quickly find.

If you really want to track `Word8#`-versus-`Word#` in the types, I guess the necessary CPP looks like this:

```
#if !MIN_VERSION_base(4,13,0)
type Word8# = Word#
wordToWord8# :: Word# -> Word8#
wordToWord8# = narrow8Word#

word8ToWord# :: Word8# -> Word#
word8ToWord# x = x
#elif !MIN_VERSION_base(4,16,0)
wordToWord8# :: Word# -> Word8#
wordToWord8# = narrowWord8#

word8ToWord# :: Word8# -> Word#
word8ToWord# = extendWord8#
#endif
```

Or the compatibility story gets easier with boxed types, with the trade-off being that we need to make sure GHC can unbox everything for us in worker/wrapper.